### DISCUSSION TITLE
Are there existing tools to aid in switching from *nix shells (zsh, bash, etc.)?

### DISCUSSION THREAD

**Discussion by jahio** [PowerShell/PowerShell#16179] [Q&A]
Are there any existing tools to help users make the switch from zsh, bash, etc. to PowerShell?

Basically, I've got a lot of customization in `~/.zshrc` going on - aliases, shell functions, environment variables, convention-over-configuration stuff that gets automatically picked up, that sorta thing. If I want to switch to PowerShell, while the environment variables can be inherited by the `pwsh` process if I launch it from _within_ an existing `zsh` process that looks at `~/.zshrc` and sets them there, that seems like fundamentally the wrong approach. And sure I could make all that stuff into shell scripts of some kind, but that's a lot of work too.

So short of rewriting all that stuff in PowerShell, is there any tools that will let `pwsh` read that file and do the same exact thing, just in a PowerShell idiomatic way without needing reliance on inheriting environment from a parent `zsh` process first?

I suspect not, but figured I'd check. And if that's indeed the case, my next question becomes pertinent: **is there any form of shell-agnostic "settings" library/tool/DSL out there** where you could, say, rename some commands, set variables/values, tell it some tasks to do to set up and/or read config on boot, etc. **where I could write out those settings and have them working/implemented directly in both `pwsh` AND other shells like `zsh` or `bash`?**

For example, let's say I want to redefine `ls` as an alias to [exa](https://github.com/ogham/exa). I'd want a DSL where I could do that, say in YAML for example:

```yaml
aliases:
  - ls:
     command: /path/to/exa --all --long --header --git
```

Then I'd need a tool to read that file and implement the `alias` in `zsh` or `bash`, or in `pwsh` define a new cmdlet or function or whatever that delegates to `exa` under the hood as defined in the config.

Does something like this already exist?

**Comment by kasini3000** [comment]
i share some tips: 

Question: How does the script read and write environment variables in [current process]?
answer:
General settings:
1 Stored in /etc/profile is the local environment variables
2 What is stored in the .bashrc in the user's home directory is the user environment variable
3 In the current process, set the environment variable, there is in the child process, but not in the parent process.
4 In the current process, display all environment variables, the command: env
5 The profile of powershell is in /user's home directory/.config/powershell/Microsoft.PowerShell_profile.ps1
Can be here, variable archives.

bash:
export A=B #setting
echo $A #read

csh, tcsh:
setenv A B #Set
echo $A #read

powershell:
$env:A ='b' #setting
$env:A #read


Q: How to create a new alias using PowerShell?
Answer: New-Alias -Name commandname -Value "command args"

q:  one click install powershell linux?
a:  https://gitee.com/chuanjiao10/kasini3000_agent_linux
